src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,180 @@
+import { Component } from "eightbittr/lib/Component";
+import { ICallbackRegister, IMod } from "modattachr/lib/IModAttachr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { INewPokemon } from "../constants/Pokemon";
+
+interface IItemProbabilities {
+    /**
+     * Name of item.
+     */
+    name: string;
+
+    /**
+     * Probability of Pokemon having item.
+     */
+    probability: number;
+}
+
+ /**
+  * Mod that randomizes items found on wild Pokemon.
+  */
+export class RandomHeldItemsMod<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> implements IMod {
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         onRandomItems: (chosenInfo: INewPokemon): INewPokemon => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const generatedNumber = this.gameStarter.numberMaker.randomReal1();
+             let counter: number = 0;
+             for (const chosenObject of this.typeItems[pokemonType]) {
+                 counter += chosenObject.probability;
+                 if (counter >= generatedNumber) {
+                     chosenInfo.item = this.gameStarter.constants.items.byName[chosenObject.name].name;
+                     break;
+                 }
+
+             }
+
+             return chosenInfo;
+         }
+     };
+
+     /**
+      * What items can be found on wild Pokemon by their primary type.
+      *
+      * @remarks No need to make probabilites add to 100.
```

Comment:
This `@remarks` seems a little vague. Can you be more specific about why?

Also, it would be adding to 1, not 100.